smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -209,16 +211,38 @@ private boolean needsInlineTimestampSchema(Context<? extends Trait> context, Mem
                 param.style("form").explode(true);
             }
 
-            // Create the appropriate schema based on the shape type.
-            Schema refSchema = context.inlineOrReferenceSchema(member);
-            QuerySchemaVisitor<T> visitor = new QuerySchemaVisitor<>(context, refSchema, member);
-            param.schema(target.accept(visitor));
+            param.schema(createQuerySchema(context, member, target));
+            result.add(param.build());
+        }
+
+        for (HttpBinding binding : httpBindingIndex.getRequestBindings(operation, HttpBinding.Location.QUERY_PARAMS)) {
+            MemberShape member = binding.getMember();
+            // To allow undefined parameters of a specific type, the name is set to `freeForm` using the `form` style.
+            ParameterObject.Builder param = ModelUtils.createParameterMember(context, member)
+                    .in("query")
+                    .name("freeForm")
+                    .style("form");
+            MapShape target = context.getModel().expectShape(member.getTarget(), MapShape.class);
+
+            // If the value of the map targets a list, the query string are repeated, so we need to "explode" them.
+            if (context.getModel().expectShape(target.getValue().getTarget()) instanceof ListShape) {
```

Comment:
`ListShape` should be `CollectionShape` to account for `set`.

Reply:
Fixed in 0c8b403.